lib/index.coffee

```diff
@@ -14,9 +15,25 @@ dockerMtimeStream()
 		current_mtimes = layer_mtimes
 
 garbageCollect = (reclaimSpace) ->
-	dokcerImageTree()
-	.then(annotateTree.bine(null, current_mtimes))
+	dockerImageTree()
+	.then(annotateTree.bind(null, current_mtimes))
 	.then(lruSort)
 	.then (candidates) ->
 		# Remove candidates until we reach `reclaimSpace` bytes
-		console.log('foobar')
+		# Candidates is a list of images, with the least recently used
+		# first in the list
+
+		# Decide on the images to remove
+		size = 0
+		remove = _.takeWhile candidates, (image) ->
+			return false if size >= reclaimSpace
+			size += image.size
+			return true
+
+		# Request deletion of each image
+		Promise.map remove, (image) ->
```

Comment:
You can also do
```coffee
	return remove
.map (image) ->
```
to do the `map` on the parent promise chain, rather than starting a new segment here